lib/datadog/kit/events.rb

```diff
@@ -7,41 +7,62 @@ module Datadog
   module Kit
     # Tracking events
     module Events
-      # Attach login event information to the trace
+      APPSEC_LOGIN_SUCCESS_EVENT = 'users.login.success'
+      APPSEC_LOGIN_FAILURE_EVENT = 'users.login.failure'
+
+      # Attach login success event information to the trace
+      #
+      # This method is experimental and may change in the future.
       #
       # @param trace [TraceOperation] Trace to attach data to.
-      # @param success [bool] Whether login operation was successful or not.
       # @param user [Hash<Symbol, String>] User information to pass to
       #   Datadog::Kit::Identity.set_user. Must contain at least :id as key.
       # @param others [Hash<String || Symbol, String>] Additional free-form
       #   event information to attach to the trace.
-      def self.track_login(trace, success:, user:, **others)
-        subtag = success ? 'success' : 'failure'
-        event = "users.login.#{subtag}"
-
-        track(event, trace, **others)
+      def self.track_login_success(trace, user:, **others)
+        track(:appsec, APPSEC_LOGIN_SUCCESS_EVENT, trace, **others)
 
         user_options = user.dup
         user_id = user.delete(:id)
 
         raise ArgumentError, 'missing required key: :user => { :id }' if user_id.nil?
```

Comment:
Minor: We could probably simplify this by doing `if user.key?(:id)` -- we're splatting the `user` in the last line so this dup/mutate ends up being kinda useless.

Reply:
Good catch

Reply:
Ah I seem to recall why I did that: static typing. `id:` being a mandatory kwarg for `set_user`, it can't be known statically that it's in the option hash, and even if it is, of which type (only the hash generic value type is known), thus typing failed.

I'll add that as an upstream issue because it *could* at least be known with a `if user.key?(:id)` check but `steep` doesn't seem to support that (maybe intentional, on grounds of concurrent mutability) and if the hash value type matches then it should be able to pass.